pyproject.toml

```diff
@@ -0,0 +1,104 @@
+[build-system]
+build-backend = "setuptools.build_meta"
+requires = ["setuptools>=61.2", "wheel"]
+
+[project]
+name = "xgi"
+dynamic = ["version"]
+description = "XGI is a Python package for higher-order networks."
+readme = "README.md"
+requires-python = ">=3.8"
+license = {text = "3-Clause BSD license"}
+authors = [
+    {name = "XGI Developers", email = "nicholas.landry@virginia.edu"}
+]
+maintainers = [
+    {name = "XGI Developers", email = "nicholas.landry@virginia.edu"}
+]
+keywords = [
+    "networks",
+    "higher-order",
+    "higher-order network",
+    "hypergraph",
+    "simplicial complex",
+    "network science",
+]
+classifiers = [
+    "Development Status :: 5 - Production/Stable",
+    "Intended Audience :: Developers",
+    "Intended Audience :: Science/Research",
+    "License :: OSI Approved :: BSD License",
+    "Operating System :: OS Independent",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: 3.12",
+    "Programming Language :: Python :: 3 :: Only",
+    "Topic :: Software Development :: Libraries :: Python Modules",
+    "Topic :: Scientific/Engineering :: Bio-Informatics",
+    "Topic :: Scientific/Engineering :: Information Analysis",
+    "Topic :: Scientific/Engineering :: Mathematics",
+    "Topic :: Scientific/Engineering :: Physics",
+]
+dependencies = [
+    "numpy>=1.20",
+    "scipy>=1.8",
+    "pandas>=1.3",
+    "networkx>=2.7",
+    "requests>=2.0",
+    "matplotlib>=3.4",
+    "seaborn>=0.10",
+]
```

Comment:
we don't have more? can we import them directly from the requirement file?

Reply:
I copied this directly from `default.txt`, so this is the accurate list of requirements. In PEP621, the goal is for all of the project info to be contained in pyproject.toml and use this file to generate all peripheral files. See for example what networkx [does](https://github.com/networkx/networkx/blob/main/tools/generate_requirements.py). In fact, I think that in the future, we might be able to remove the requirements files all together since I can't think of an example where one wishes to only import the dependencies and not the package itself. But that's a future problem, I think.

Reply:
Ah I see so this should ultimately replace the requirement files.

Reply:
Yep!